main.go

```diff
@@ -132,6 +171,60 @@ func main() {
 	}
 }
 
+func deleteAction(ctx context.Context, clientSet client.Client, hr *fluxhelmv2beta1.HelmRelease, timeout, interval time.Duration) {
+	instance := &fluxhelmv2beta1.HelmRelease{}
+	key := client.ObjectKey{
+		Name:      hr.Name,
+		Namespace: hr.Namespace,
+	}
+	if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+		log.Fatalf("Failed to get instance of the helmrelease with %v", err)
+	} else if err != nil {
+		// This means that the object was not found
+		log.Errorf("Did not find the helm release: %v", hr.Name)
+	} else {
+		instance.Annotations = hr.Annotations
+		instance.Labels = hr.Labels
+		instance.Spec = hr.Spec
+
+		log.Infof("Found the helm release: %v, deleting the release...", hr.Name)
+		// Delete the HelmRelease
+		retry(ctx, func() error { return clientSet.Delete(ctx, instance) }, interval)
+	}
+
+	if err := PollHelmReleaseDeleted(ctx, clientSet, hr, timeout); err != nil {
+		log.Errorf("Failed to poll the deletion of the helm release with %v", err)
+		if err := clientSet.Get(ctx, key, instance); client.IgnoreNotFound(err) != nil {
+			log.Fatalf("Failed to get instance of the helmrelease with %v", err)
+		} else if err != nil {
+			log.Errorf("Did not find the helm release: %v, the object is deleted", hr.Name)
+		} else {
+			instance.Annotations = hr.Annotations
+			instance.Labels = hr.Labels
+			instance.Spec = hr.Spec
+
+			log.Infof("Removing the helm release finalizer: %v...", fluxhelmv2beta1.HelmReleaseFinalizer)
+			controllerutil.RemoveFinalizer(instance, fluxhelmv2beta1.HelmReleaseFinalizer)
+			if err := clientSet.Update(ctx, instance); err != nil {
+				log.Fatalf("Failed to update the helmrelease with %v", err)
+			}
+			if err := PollHelmReleaseDeleted(ctx, clientSet, hr, timeout); err != nil {
```

Comment:
Can you explain what the intended behavior you are trying to capture here is? My thought is that we should attempt to wait for the HelmRelease to remove its own finalizer by polling for the helm release to be gone from the cluster. If we timeout on the polling, then we should remove the finalizer ourself and exit. Let me know if my understanding is wrong here

Reply:
Yes that is what we try to do on timeout - i.e
1. Remove the finalizer from the `HelmRelease` and have it cleaned up
2. (for safety) Since we are interrupting the helm-controller cleanup by force deleting the finalizer, we make a last attempt to delete the underlying `helm release` in case helm-controller left it behind